src/static/scripts/ajaxify.js

```diff
@@ -0,0 +1,44 @@
+/* global ajax */
+/* exported loadContent */
+
+var loadContent = (function() {
+    var pageCache = {};
+
+    function createRequest(url, callback) {
+        function errorCallback() {
+            console.log('error');
+        }
+        function httpErrorCallback() {
+            console.log('http error');
+        }
+        function successCallback() {
+            console.log('success');
+            callback(this.responseText);
+        }
+        ajax(url)
+            .error(errorCallback)
+            .httpError(httpErrorCallback)
+            .success(successCallback)
+            .requestHeader('X-Ajax-Request', true)
+            .send();
+    }
+
+    function fillContent(element, data) {
+        element.html(data);
+    }
+
+    return function(url, element) {
+        console.log(pageCache);
+        if (pageCache.hasOwnProperty(url)) {
```

Comment:
This could be simplified, if you wish so. Only check if the `url` is not in the cache, and if so, load it. Finally, always `fillContent` using the cache.